sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/generators.scala

```diff
@@ -444,7 +444,8 @@ case class Inline(child: Expression) extends UnaryExpression with CollectionGene
   }
 
   override def elementSchema: StructType = child.dataType match {
-    case ArrayType(st: StructType, _) => st
+    case ArrayType(st: StructType, false) => st
+    case ArrayType(st: StructType, true) => st.asNullable
```

Comment:
qq: does Hive also interprets `null` as `Row(null, ...)` in `inline` too? If that's the case, the change seems fine.

Reply:
Yes, with Hive 2.3.9:
```
Beeline version 2.3.9 by Apache Hive
0: jdbc:hive2://localhost:10000> select inline(array(named_struct('a', 1, 'b', 2), null));
+-------+-------+
|   a   |   b   |
+-------+-------+
| 1     | 2     |
| NULL  | NULL  |
+-------+-------+
2 rows selected (1.355 seconds)
0: jdbc:hive2://localhost:10000> 
```